tests/endpoints_test.rs

```diff
@@ -0,0 +1,72 @@
+#[path = "common.rs"]
+mod common;
+
+mod tests {
+    use crate::common::{build_app, initialize_logging};
+    use axum::{
+        body::{to_bytes, Body},
+        http::Request,
+    };
+    use blockfrost_platform::api::root::RootResponse;
+    use pretty_assertions::assert_eq;
+    use reqwest::{Method, StatusCode};
+    use tower::ServiceExt;
+
+    // Test: `/` route correct response
+    #[tokio::test]
+    async fn test_root_route() {
+        initialize_logging();
+
+        let (app, _handle) = build_app().await.expect("Failed to build the application");
+
+        let response = app
+            .oneshot(Request::builder().uri("/").body(Body::empty()).unwrap())
+            .await
+            .expect("Request to root route failed");
+
+        assert_eq!(response.status(), StatusCode::OK);
+
+        let body_bytes = to_bytes(response.into_body(), usize::MAX)
+            .await
+            .expect("Failed to read response body");
+        let root_response: RootResponse =
+            serde_json::from_slice(&body_bytes).expect("Response body is not valid JSON");
+
+        assert!(root_response.errors.is_empty());
+        assert_eq!(root_response.name, "blockfrost-platform");
+        assert!(root_response.healthy);
+        assert_eq!(root_response.node_info.sync_progress, 100.0);
```

Comment:
I started writing that it's sometimes a little less, but you rounded to 2 decimal places above! :bulb: